gradle/build-logic/check-dexer/src/main/kotlin/DexerUtils.kt

```diff
@@ -0,0 +1,118 @@
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.attributes.Usage
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.JavaExec
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+import org.gradle.process.CommandLineArgumentProvider
+import java.io.File
+import java.io.OutputStream
+
+fun configureDexerForProject(project: Project, dexerPreCheckTask: Provider<Task>) {
+
+    project.repositories.apply {
+        google()
+        mavenCentral()
+    }
+
+    configureR8(project)
```

Comment:
Nitpick: I would use twice the same naming:
```suggestion
    configureR8Configuration(project)
```

or twice without Configuration

Reply:
Agree, but I preferred `configureR8` and `configureD8` as consistent naming. 
Somehow `configure..Configuration` sounds redundant :)